### PR TITLE
Fixes #121 log4j config

### LINKED ISSUE #121: migrator: puppetize log4j configuration
This would be a follow up to https://github.com/lightblue-platform/lightblue-migrator/issues/111

### PR DESCRIPTION
**Pull request by dcrissman**
Adds the ability for puppet to configure log4j outside of the embedded default configuration.

This is also a large refactor of existing functionality to ensure that everything acts consistently.

### CODE REVIEW

File: manifests/application/migrator/log4j.pp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  $log_level = 'INFO',
+  $log_file_name = 'migrator.log',
+  $log_max_file_size = '1MB',
+  $log_max_backups_to_keep = '5',
+  $log_pattern = '%d [%t] %c %p - %m%n',
+  $config_dir,
+  $log_dir,
+  $owner,
+  $group,
+  $service_name
```
Comment (jewzaam) [inline manifests/application/migrator/log4j.pp]: Regarding defaults for lightblue::application::migrator::log4j I recommend:
* $log_max_file_size = '10MB'
* $log_max_backups_to_keep = '50'
* $log_pattern = '%d [%t] %-5p [%c] %m%n'

These are consistent with other logging we have for other systems.

File: manifests/application/migrator/log4j.pp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  $log4j_config_file = "$config_dir/log4j.properties"
+  $log_file = "$log_dir/$log_file_name"
+
+  file{ $log4j_config_file:
+    ensure  => 'file',
+    mode    => '0644',
+    owner   => $owner,
+    group   => $group,
+    content => template('lightblue/application/migrator/log4j.properties.erb'),
+    notify  => [Service[$service_name]]
```
Comment (jewzaam) [inline manifests/application/migrator/log4j.pp]: I recommend taking out notifying the service on logging changes.  Either:
1. make restarting the service a manual effort
2. update service to pick up logging dynamically.

I'm for the second option, created https://github.com/lightblue-platform/lightblue-migrator/issues/128

Comment (jewzaam) [inline manifests/application/migrator/log4j.pp]: Shouldn't params with default values be listed before those without defaults?  I think our linting will complain.

File: manifests/application/migrator.pp [hunk 3]
```diff
@@ -22,31 +26,31 @@
...
+# $primary_client_ca_file_path       - (optional) ca file path
 # $primary_client_ca_source          - (optional) source for ca file.
-# $primary_client_cert_file_path     - see lightblue::client::configure
+# $primary_client_cert_file_path     - (optional) cert file path
 # $primary_client_cert_source        - (optional) source for cert file.
 # $primary_client_cert_password      - see lightblue::client::configure
 # $primary_client_cert_alias         - see lightblue::client::configure
 #
-# $source_config                     - Source lightblue client configuration file. Defaults to $config_file.
+# $is_source_same_as_primary         - Indicates if the source lightblue instance is the same as the primary. Defaults to true.
```
Comment (jewzaam) [inline manifests/application/migrator.pp]: Two things:
1. this has nothing to do with the context of the PR
2. why not just check if source config = primary config, like it was doing before?

**Comment by jewzaam** [comment]
Moved back to 'in progress'.  This PR contains many changes around how clients are configured.  Please pull those out to a separate PR or include additional context in the PR description.

**Comment by dcrissman** [comment]
@jewzaam You're right, I have too much going on here. I am working to split this into smaller PRs.

**Comment by dcrissman** [comment]
Per a conversation with @jewzaam I will re-open this PR.

The migrator puppet script up until this point has depended on the client code to provide the absolute paths for all config files. This is the first time (aside from the service script) puppet is actually landing a configuration file for the migrator. This requires a conf directory to place the log4j.properties file in. 

Now with landing that file, it now seems inconsistent that other configuration files are not able to be landed as well or in the same directory. That led to a much larger refactor to empower puppet to be able to land everything in a consist directory structure. It is however, still possible to use your own lightblue config files if that is desired.

File: manifests/application/migrator.pp (hunk 3 above)
Comment (dcrissman) [inline manifests/application/migrator.pp]: Previously the assumption was made that the primary, source, and destination config files would always be provided if they were being used. Now it is more likely that puppet will be generating and landing those configuration files.

File: manifests/application/migrator.pp [hunk 4]
```diff
@@ -66,9 +70,13 @@
...
     $migrator_version = 'latest',
     $jsvc_version = 'latest',
+    $migrator_home_dir = '/usr/share/migrator',
+    $migrator_config_dir = '/etc/migrator',
+    $migrator_log_dir = '/var/log/migrator',
+    $generate_log4j = false,
     $java_home = undef,
-    $jar_path = '/var/lib/jbossas/standalone/deployments/lightblue-migrator-consistency-checker-*.jar',
-    $service_log_file = 'console.log',
+    $jar_path = '/usr/share/jbossas/standalone/deployments/lightblue-migrator-consistency-checker-*.jar',
```
Comment (jewzaam) [inline manifests/application/migrator.pp]: Why is this jar deployed in a jboss dir?  There is no jboss required for migrator.

File: manifests/application/migrator.pp [hunk 5]
```diff
@@ -116,168 +123,326 @@
...
     }
 
-    if($source_config_file != $primary_config_file){
+    #configure source lightblue instance
+    if($is_source_same_as_primary){
+      $source_config_file = $primary_config_file
+    }
+    else {
       if !$source_client_metadata_uri or !$source_client_data_uri {
         fail('If defining a source_config, then you must also define data and metadata urls for it.')
```
Comment (jewzaam) [inline manifests/application/migrator.pp]: For this error would be nice to have explicit note of what params are not set

Comment (jewzaam) [inline manifests/application/migrator.pp]: Ignore this, can tackle with https://github.com/lightblue-platform/lightblue-puppet/issues/125

File: manifests/application/migrator.pp (hunk 4 above)
Comment (jewzaam) [inline manifests/application/migrator.pp]: https://github.com/lightblue-platform/lightblue-puppet/issues/126

Comment (dcrissman) [inline manifests/application/migrator.pp]: That is true. It has always been this way, I am not sure of the original reasoning.

**Pull request merged**